File: CodeForces/Round 596/B.cpp
    #include <bits/stdc++.h>
    using namespace std;

    #define FOR(i,a,b) for (int i = (a); i < (b); ++i)
    #define F0R(i,a) FOR(i,0,a)
    #define ROF(i,a,b) for (int i = (b)-1; i >= (a); --i)
    #define R0F(i,a) ROF(i,0,a)

    using ll = long long;
    ll INF = 1LL << 60;
    
    using vc = vector<char>;
    using vi = vector<int>;

    void solve() {
        int n, k, d; cin >> n >> k >> d;
        unordered_map<int, int> occ;
        vi arr (n);
        F0R(i, n) cin >> arr[i];

        int types = 0;
        F0R(i, d) {
            if (occ[arr[i]] == 0) ++types;
            ++occ[arr[i]];
        }
        int best = types;

        int lo = 0, hi = d;
        for (; hi < n; ++lo, ++hi) {
            best = min(best, types);
            --occ[arr[lo]];
            if (occ[arr[lo]] == 0) --types;
            if (occ[arr[hi]] == 0) ++types;
            ++occ[arr[hi]   ];
            best = min(best, types);

        }
        
        cout << best << endl;
    }
    int main() {
        ios::sync_with_stdio(0), cin.tie(0), cout.tie(0);
        int t; cin >> t;
        F0R(i, t) solve();
    }

File: CodeForces/Round 596/B.py
T = int(input())
for i in range(T):
    N, K, D = map(int, input().split())
    shows = list(map(int, input().split()))

    count = [0 for i in range(K + 1)]
    types = 1

    best = float('inf')
    lo = 0
    hi = 0
    count[shows[0]] += 1
    while hi < N:
        
        
        if hi - lo + 1 < D:
            hi += 1
            
            try:
                if count[shows[hi]] == 0:
                    types += 1
                count[shows[hi]] += 1
            except:
                pass
        else:
            best = min(best, types)
            
            count[shows[lo]] -= 1
            if count[shows[lo]] == 0:
                types -= 1

            lo += 1

    print(best)
